std/functional.d

```diff
@@ -1562,3 +1565,45 @@ template forward(args...)
     int value = 3;
     auto x2 = bar(value); // case of OK
 }
+
+/**
+$(D auto apply(Tup)(Tup t)
+    if (isTuple!Tup && Parameters!callable.length == Tup.length))
+
+Applies the items of tuple `t` as arguments to the function `callable`.
+
+Returns:
+    The return value of the function `callable` for the arguments
+    represented by the items of `t`.
+
+*/
+template apply(alias callable)
+{
+    import std.typecons : isTuple;
+    auto apply(Tup)(Tup t)
+    if (isTuple!Tup && Parameters!callable.length == Tup.length)
+    {
+        return callable(t.tupleof);
+    }
+}
+
+///
+@safe unittest
+{
+    import std.range : zip;
+    import std.algorithm.comparison : equal;
+    import std.algorithm.iteration : map;
+    import std.typecons : tuple;
+
+    auto tup = tuple('a', 10);
+    auto a1 = "abcd"d;
+    auto a2 = [10, 20, 30, 40];
+    auto a3 = [100, 200, 300, 400];
+
+    int foo(dchar a, int b) { return a + b; }
+    assert(tup.apply!foo() == 107);
+    assert(zip(a1, a2).map!(apply!foo).equal([107, 118, 129, 140]));
+
+    int foo2(dchar a, int b, int c) { return a + b + c; }
+    assert(zip(a1, a2, a3).map!(apply!foo2).equal([207, 318, 429, 540]));
+}
```

Comment:
Please also test overload sets - two or more functions with the same name and different parameters.